File: UniversityApi.API/DataBase/Configuration/CourseConfiguration.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System.Reflection.Emit;
using UniversityApi.API.DataBase.Entities;

namespace UniversityApi.API.DataBase.Configuration
{
    public class CourseConfiguration : IEntityTypeConfiguration<EntityCourse>
    {
        public void Configure(EntityTypeBuilder<EntityCourse> builder)
        {
            builder.HasData(
                new EntityCourse
                {
                    EntityCourseID = Guid.NewGuid(),
                    CourseCode = "C01",
                    Name = "Databases Seed",
                    ECTS = 2,
                    IsFinishedWithExam = false,
                },
                new EntityCourse
                {
                    EntityCourseID = Guid.NewGuid(),
                    CourseCode = "C02",
                    Name = "Algorithms Seed",
                    ECTS = 3,
                    IsFinishedWithExam = true,
                },
                new EntityCourse
                {
                    EntityCourseID = Guid.NewGuid(),
                    CourseCode = "C03",
                    Name = "Computer science seed",
                    ECTS = 4,
                    IsFinishedWithExam = true,
                }); ;
        }
    }
}

File: FsUniversityApi/Database/Configurations/CourseConfiguration.fs
namespace FsUniversityApi.Database.Configurations

//open FsUniversityApi.Database.Entities.Course
//open Microsoft.EntityFrameworkCore
//open Microsoft.EntityFrameworkCore.Metadata.Builders
//open System

//type CourseConfiguration() =
//    interface IEntityTypeConfiguration<Course> with
//        member this.Configure(builder : EntityTypeBuilder<Course>) =
//            builder.HasData(
//                //{ CourseId = Guid.NewGuid(); CourseCode = "C01"; CourseName = "Java course"; IsFinishedWithExam = false; ECTS = 2; SoftDeleted = false },
//                //{ CourseId = Guid.NewGuid(); CourseCode = "C02"; CourseName = "Databases"; IsFinishedWithExam = true; ECTS = 4; SoftDeleted = false },
//                //{ CourseId = Guid.NewGuid(); CourseCode = "C03"; CourseName = "Algorithms"; IsFinishedWithExam = true; ECTS = 5; SoftDeleted = false }
//                ) |> ignore